eppo_core/src/events/event_dispatcher.rs

```diff
@@ -51,7 +54,10 @@ impl<'a> EventDispatcher {
         let batch_size = config.batch_size;
         loop {
             let mut batch_queue: Vec<Event> = Vec::with_capacity(batch_size);
-            let ingestion_url = config.ingestion_url.clone();
+            let event_delivery = EventDelivery::new(
+                config.sdk_key.clone(),
+                config.ingestion_url.clone()
+            );
```

Comment:
minor(perf): this creates a new reqwest client on each batch. Reqwest recommends against that because the client manages some state (e.g., concurrent requests and reusing an active connection to the server)

Reply:
<img width="679" alt="image" src="https://github.com/user-attachments/assets/74345492-36bf-439d-82eb-c2576fd3f21e" />
i couldn't figure out how to reuse without running into this borrow checker error